spec/test_app/app/controllers/redirects_controller.rb

```diff
@@ -0,0 +1,8 @@
+class RedirectsController < ApplicationController
```

Comment:
We don't need to test this by adding a new controller to the test app, we can just test `GovukPersonalisation::Redirect` directly - the test app is because one of the things to test is a Rails concern, which can't really be tested outside the context of an app.

Reply:
We did it this way because we weren't sure how to follow the redirects in the method without having the app and the Rspec Rails helpers. Have you got an example of testing redirects in a standalone module?

Reply:
We use `redirect_to` [on](https://github.com/alphagov/govuk_personalisation/pull/19/files#diff-a05499a5b5e5dcff58d3227904777f2aefa2c9f0264f7ca3b52b80e53eca5e6bR5) a [couple of](https://github.com/alphagov/govuk_personalisation/pull/19/files#diff-a05499a5b5e5dcff58d3227904777f2aefa2c9f0264f7ca3b52b80e53eca5e6bR11) lines.

I'd figured that because that was an [ActiveController method ](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_to)we'd run into problems trying to test this as a module outside of rails.

That necessitatest the test app, no?

We talked about simplifying this so it just returns a string and the redirects happen outside the module.
But the only other consumer of this would be changing `href`s on the page. And those it looks like we'll need to do in JS for session reasons.

So seemed less nice to not use a redirect here.

Reply:
Sorry, I missed that the method was redirecting rather than just returning the URL, you're right.